test/split.test.ts

```diff
@@ -0,0 +1,244 @@
+        type: 7 % len,
+        claimableAmount: BigNumber.from(7),
+        index: 7,
+        proof,
+      },
+      {
+        account: "0x" + "4".repeat(40),
+        type: 4 % len,
+        claimableAmount: BigNumber.from(4),
+        index: 4,
+        proof,
+      },
+      {
+        account: "0x" + "6".repeat(40),
+        type: 6 % len,
+        claimableAmount: BigNumber.from(6),
+        index: 6,
+        proof,
+      },
+    ];
+    const size = 3;
+    const result = [...splitClaims(claims, size)];
+    const mapping = result.map(([pair]) => pair);
+    const chunks = result.map(([, chunk]) => chunk);
+    expect(mapping).to.deep.equal([
+      ["0x" + "1".repeat(40), "0x" + "3".repeat(40)],
+      ["0x" + "4".repeat(40), "0x" + "6".repeat(40)],
+      ["0x" + "7".repeat(40), "0x" + "7".repeat(40)],
+    ]);
+    const expected: ClaimChunk[] = [
+      {
+        ["0x" + "1".repeat(40)]: [
+          { type: ClaimType[1 % len], amount: "1", index: 1, proof },
+        ],
+        ["0x" + "2".repeat(40)]: [
+          { type: ClaimType[2 % len], amount: "2", index: 2, proof },
+        ],
+        ["0x" + "3".repeat(40)]: [
+          { type: ClaimType[3 % len], amount: "3", index: 3, proof },
+        ],
+      },
+      {
+        ["0x" + "4".repeat(40)]: [
+          { type: ClaimType[4 % len], amount: "4", index: 4, proof },
+        ],
+        ["0x" + "5".repeat(40)]: [
+          { type: ClaimType[5 % len], amount: "5", index: 5, proof },
+        ],
+        ["0x" + "6".repeat(40)]: [
+          { type: ClaimType[6 % len], amount: "6", index: 6, proof },
+        ],
+      },
+      {
+        ["0x" + "7".repeat(40)]: [
+          { type: ClaimType[7 % len], amount: "7", index: 7, proof },
+        ],
+      },
+    ];
+    expect(chunks).to.deep.equal(expected);
+  });
+
+  it("joins claims for the same user", async function () {
+    const claims: ProvenClaim[] = [
+      {
+        account: "0x" + "1".repeat(40),
+        type: ClaimType.Advisor,
+        claimableAmount: BigNumber.from(42),
+        index: 42,
+        proof,
+      },
+      {
+        account: "0x" + "1".repeat(40),
+        type: ClaimType.Airdrop,
+        claimableAmount: BigNumber.from(4242),
+        index: 4242,
+        proof,
+      },
+      {
+        account: "0x" + "2".repeat(40),
+        type: ClaimType.GnoOption,
+        claimableAmount: BigNumber.from(1337),
+        index: 1337,
+        proof,
+      },
+    ];
+    const size = 3;
+    const result = [...splitClaims(claims, size)];
+    const mapping = result.map(([pair]) => pair);
+    const chunks = result.map(([, chunk]) => chunk);
+    expect(mapping).to.deep.equal([
+      ["0x" + "1".repeat(40), "0x" + "2".repeat(40)],
+    ]);
+    expect(chunks).to.deep.equal([
+      {
+        ["0x" + "1".repeat(40)]: [
+          { type: "Advisor", amount: "42", index: 42, proof },
+          { type: "Airdrop", amount: "4242", index: 4242, proof },
```

Comment:
What if the same user appears twice with the same category? Of course I would not be sending that information to you. Just asking

Reply:
Basically the same claim counts twice. Claims in the contract are identified by ids, and as long as the ID is different all the rest can be the same. If gas was cheap, it would have been nice to have many claims depending on how they were obtained (e.g., one claim each for trading, community, poap, ...).
Since it's a good question I also added a [test](https://github.com/gnosis/cow-token/pull/43) for it.